src/components/Grid/snapping.js

```diff
@@ -80,9 +80,20 @@ export function findClosestEdge(edges, cursor) {
     withDistance = edges.map(e => ({
       ...e,
       ...pointDistanceToSegment(cursor, { start: e.v1, end: e.v2 }),
-    })),
-    closestEdge = _.minBy(withDistance, 'dist');
-  return closestEdge;
+    }));
+
+  let ret = undefined;
+  let smallest = NaN;
+  withDistance.forEach((d) => {
+    // The conditional is intentionally convoluted - simplifying it to just '' will have issues because it starts as NaN
+    // All comparisons to NaN return false, so for the first iteration it has to negate it
+    if (!(d.dist >= smallest)) {
+      smallest = d.dist;
+      ret = d;
+    }
+  });
```

Comment:
would this be clearer if we started at `Infinity` rather than NaN? Is this a meaningful speed-up over `_.minBy`?

Reply:
It's about ~5%, so not the biggest speed up.